### DISCUSSION TITLE
Can't dynamically convert SetUp URL for Github Apps?

### DISCUSSION THREAD

**Discussion by oct-sky-out** [community/community#109542] [Apps, API and Webhooks]
### Select Topic Area

Question

### Body

In the business logic we are developing, I currently want to proceed through the post installation Setup URL without adopting OAuth authentication after successful Github Apps installation to continue using the service.

In this case, I want to send a redirect from the Setup URL to a dynamic URL, is there any solution?

Here is an example of the business logic I want to create.
1. user navigates to /post/1/create
2. the condition for the user to stay on the /post/1/create page is whether or not they have Github Apps installed.
3. if not installed, navigate to Github Apps install page/complete installation (ex. Setup URL is set http://example.com/post/:id/create)
4. recognize the Setup URL setting of Github Apps, Post installation and redirect to /post/1/create via dynamic URL conversion

**Comment by atharva5683** [comment]
What do people think to switch to https://pypi.org/project/setuptools-scm/, which generates the package version dynamically from the SCM, rather than having it hardcoded as we do today (in two places to make things worse)? virtualenv does this for example https://github.com/pypa/virtualenv/blob/main/setup.py#L7-L11
